trace/http.go

```diff
@@ -2,19 +2,46 @@ package trace
 
 import "net/http"
 
-func Get(url string, parent *http.Request) (resp *http.Response, err error) {
-	c := &http.Client{}
-	req, err := http.NewRequest("GET", url, nil)
-	if err != nil {
-		return nil, err
+// DefaultCollector is a MemoryCollector. Trace defaults back to DefaultCollector if none given.
+var DefaultCollector = NewMemoryCollector()
+
+// Trace is used to add incoming Trace functionality.
+type Trace struct {
+	collector Collector
+}
+
+func (t *Trace) process(rw http.ResponseWriter, r *http.Request, f http.HandlerFunc) {
```

Comment:
It would be cleaner and more general to take an `http.Handler` instead of an `http.HandlerFunc`.